### PR TITLE
11-Layer Int6 + WD=0.04 + SWA + FA3 (val_bpb: 1.1318)

### PR DESCRIPTION
**Pull request by jfprincz**
## 11-Layer Int6 + WD=0.04 + SWA + FA3 (val_bpb: 1.1318)

**val_bpb: 1.1318** (sliding window, stride=64) | **15.7 MB** | 8xH100 SXM, 600s

### Progress from prior submissions

| | [PR #70](https://github.com/openai/parameter-golf/pull/70) | [PR #164](https://github.com/openai/parameter-golf/pull/164) | This | Delta vs #164 |
|---|---|---|---|---|
| **val_bpb (sliding)** | 1.1659 (s256) | 1.1524 (s256) | **1.1318 (s64)** | **-0.0206** |
| Layers | 9 | 9 | 11 | +2 |
| Params | 21.8M | 22.4M | 26.8M | +4.4M |
| Artifact | 14.9 MB | 15.4 MB | 15.7 MB | +0.3 MB |
| Steps (600s) | 12,485 | 8,390 | 7,412 | -978 |
| Step time (8xH100) | 48ms | 68ms | 81ms | +13ms |

Two extra layers compensate for fewer steps. Weight decay 0.04 (Muon + AdamW) keeps weights quantization-friendly under int6. Sliding window now at stride=64.

### Key additions over PR #164

| Change | Impact |
|--------|--------|
| **11 layers** (was 9) | +4.4M params, major capacity gain, funded by int6 headroom |
| **Weight decay 0.04** | Muon WD + AdamW WD, smaller weights improve int6 compression |
| **SWA** (~8 checkpoint avg) | Smoothed weights during warmdown |
| **Eval stride=64** (was 256) | Near-full context for every scored token |
| **Bigram 2048 buckets** (was 4096) | Saves ~300 KB artifact, negligible BPB cost |

Everything else from PR #164 carries forward: OrthoInit + muP, 3x MLP, int6 mixed quant + zstd-22, SmearGate, BigramHash, FA3, seq 2048, tuned Muon.

### Results

| Metric | Value |
|--------|-------|
| Pre-quant val_bpb | 1.1432 |
| Int6 roundtrip val_bpb | 1.1543 |
| **Int6 sliding val_bpb (s64)** | **1.1318** |
| Steps completed (600s cap) | 7,412 |
| Step time | 81ms |
| Model params | 26,829,913 |
| Artifact size | 15,689,380 bytes |

### Reproducibility (3 seeds)

| Seed | Steps | Sliding s64 | Artifact |
|------|-------|-------------|----------|
| **1337** | **7,412** | **1.1318** | **15.69 MB** |
| 42 | 7,407 | 1.1335 | 15.70 MB |
| 2025 | 7,412 | 1.1324 | 15.69 MB |

Mean: **1.1326** | Variance: 0.0017 | Submitted: seed 1337

### Run command

```bash
NUM_LAYERS=11 BIGRAM_VOCAB_SIZE=2048 \
MUON_WD=0.04 ADAM_WD=0.04 \
MATRIX_LR=0.025 SCALAR_LR=0.025 TIED_EMBED_LR=0.035 \
MUON_MOMENTUM=0.99 MUON_MOMENTUM_WARMUP_START=0.92 \
MUON_MOMENTUM_WARMUP_STEPS=1500 WARMDOWN_ITERS=3000 \
ITERATIONS=9000 MAX_WALLCLOCK_SECONDS=600 EVAL_STRIDE=64 \
torchrun --standalone --nproc_per_node=8 train_gpt.py
```

### CODE REVIEW

**Comment by MatoTeziTanka** [comment]
### Community Review — 11-Layer Int6 + WD=0.04 + SWA + FA3 (val_bpb: 1.1318)

**BPB:** 1.1318 | **Compliance:** LOOKS CLEAN — pure-neural submission, no TTT/SLOT/n-gram-cache

**What I found in the code** (head SHA `372bddea57f4`, file `records/track_10min_16mb/2026-03-20_11L_Int6_MLP3x_WD04_SmearBigram2k_1.1318/train_gpt.py`):

Static code review found no TTT adaptation function, no SLOT optimization loop, no n-gram-cache class, and no pre-quant val-token fine-tune. The eval path uses the standard sliding-window stride-64 pattern. The submission is a pure-neural architecture iteration on the standard SP1024/SP4096/SP8192 baseline.

**CPU smoke test (CT2038 proteus-engine, 2026-04-11):** import OK in 0.04s, dim=512, layers=9, vocab=1024, code=64415 B, SMOKE_TEST_PASS

**Verdict:** LOOKS CLEAN.

**Recommendation to @cocohearts @valerio-oai @0hq @yuzhougu-oai @notapplica:** **MERGE** pending the usual record-track checks (3-seed validation, under-16MB artifact cap, ≤600s train + ≤600s eval on 8×H100 SXM). No compliance flags from the classification pass — this looks like a clean pure-neural iteration on the standard baseline.

*Auto-classification caveat:* this review was drafted by the AST-based classifier. If there's a non-standard eval mechanism (logit postprocessing, hedge mixing, etc.) that I missed because it's factored into a helper file or a non-standard function name, please flag it and I'll re-run the audit manually.

---
*Reviewed by [@MatoTeziTanka](https://github.com/MatoTeziTanka) — [The Agora](https://matotezitanka.github.io/parameter-golf). CPU smoke test (CT2038 proteus-engine, 2026-04-11): import OK in 0.04s, dim=512, layers=9, vocab=1024, code=64415 B, SMOKE_TEST_PASS. Classification via deterministic AST-based `classify_prs.py` (pattern bank derived from ~65 manually-reviewed PRs earlier in the 2026-04-11 sweep). This review was auto-drafted from a template and spot-checked before posting — if the template misread your code, please call it out so I can iterate the classifier.*
